rfc/0033-quilt-mappings-and-clean-room.md

```diff
@@ -0,0 +1,97 @@
+
+With Mojang's mappings (Mojmap) quickly becoming the de-facto mappings standard throughout the modding ecosystem, and MCP mappings effectively dead and gone, it no longer makes sense to maintain the purity of Yarn. This is especially true given the original motivations behind Yarn's cleanroom approach, the questionably impure approach taken by some of the upstream maintainers, and the lack of formal accountability for its contributors.
+
+This RFC proposes that non-Yarn mappings should be allowed to be referenced in all Quilt community and development spaces, with the goal of creating a more inclusive support ecosystem and raising the quality of Quilt's fork of Yarn.
+
+As Quilt's Yarn fork will be significantly different from upstream Yarn, this RFC also proposes changing its name to Quilt Mappings (QM).
+
+Any discussions related to migrating existing Quilt projects to other mappings standards are out of scope of this RFC, and there is - as of this writing - no intention to do so.
+
+## Motivation
+
+In order to understand the motivation for this RFC, it's important to look at how we got here. A lot has changed since Yarn was concieved in 2016, and the approach it takes isn't really relevant in today's modding landscape.
+
+### Background
+
+Yarn was originally created on August 12, 2016. It was created for a project called Open Mod Loader, the first iteration of what would be known as Fabric.
+
+It would remain in relative obscurity until October of 2018, when Yarn began tracking early snapshots for Minecraft 1.14.
+
+### Rift
+
+During the 1.13 development cycle, Forge chose to rewrite most of their scaffolding, and a stable Forge build didn't appear until Minecraft 1.14. Rift was created in response to this, as a lightweight mod loader that could do basic coremodding on Minecraft 1.13.
+
+However, Rift made use of parts of the Forge toolchain without permission. Forge declined to grant this permission to them after the fact, and Rift-related discussions were effectively banned in most modding-related communities.
+
+### Yarn's Motivation
+
+Due to the strict licensing of some parts of the Forge toolchain at the time, Fabric needed to avoid making use of Forge projects such as MCP, and thus avoid making the same mistakes that Rift did. In order to achieve this, Yarn was licensed under the [Creative Commons Zero License](https://creativecommons.org/share-your-work/public-domain/cc0/) (CC0), a public domain dedication.
+
+Making use of the CC0 in this way meant that all names in Yarn needed to be original contributions - it would have been unacceptable for a name to make it in from another mappings project, unless that project was also licensed under the CC0. For this reason, Yarn needed to take a cleanroom approach, rejecting contributions that added names from other mappings projects, and banning discussions of names from non-CC0 projects in related community spaces.
```

Comment:
I missed that on my first read, sorry about that:

> Making use of the CC0 in this way meant that all names in Yarn needed to be original contributions
This paragraph again reads as if those restrictions exist because CC0 was applied. This is not true. The requirement for "original contributions" comes strictly from the informally required clean room, not from the CC0 license.
So a more correct sentence would then be:
> Additionally, in order to protect against claims of stealing from other projects, al names in Yarn needed to be original contributions - it would have been [...]
I don't know if that was Yarn's reasoning or not, but either way the original statement as it is here isn't correct. CC0 doesn't necessitate only original contributions.

Reply:
```suggestion
Additionally, in order to protect against claims of stealing from other projects, all names in Yarn needed to be original contributions - as it would have been in violation of the CC0 license to use non-CC0 names in a CC0 project. For this reason, Yarn needed to take a cleanroom approach, rejecting contributions that added names from other mappings projects, and banning discussions of names from non-CC0 projects in related community spaces.
```
@CheaterCodes

Reply:
Uhm... No?
Isn't this the exact same argument again?
> as it would have been in violation of the CC0 license to use non-CC0 names in a CC0 project.
It does *not* violate the CC0 license to use non-CC0 names in a CC0 project, as far as I understand.
I mean, that's *exactly* what we're planning to do.

Reply:
```suggestion
Additionally, in order to protect against claims of stealing from other projects, all names in Yarn needed to be original contributions. For this reason, Yarn needed to take a cleanroom approach, rejecting contributions that added names from other mappings projects, and banning discussions of names from non-CC0 projects in related community spaces.
```
how about this

Reply:
@CheaterCodes

Reply:
That's fine, yeah.